src/AutoRest.CSharp/LowLevel/Generation/LowLevelClientWriter.cs

```diff
@@ -393,6 +367,77 @@ private static void WriteLongRunningOperationMethod(CodeWriter writer, LowLevelL
             writer.Line();
         }
 
+        private void WriteSubClientFactoryMethod(CodeWriter writer, BuildContext context, LowLevelRestClient parentClient, LowLevelRestClient[] subClients)
+        {
+            var factoryMethods = new List<(FieldDeclaration?, MethodSignature, List<Reference>)>();
+            foreach (var subClient in subClients)
+            {
+                var methodParameters = new List<Parameter>();
+                var constructorCallParameters = new List<Reference>();
+
+                foreach (var parameter in subClient.SubClientInternalConstructor.Parameters)
+                {
+                    var field = parentClient.GetFieldReferenceByParameter(parameter);
+                    if (field == null)
+                    {
+                        methodParameters.Add(parameter);
+                        constructorCallParameters.Add(parameter);
+                    }
+                    else
+                    {
+                        constructorCallParameters.Add(new Reference(field.Name, field.Type));
+                    }
+                }
+
+                var subClientName = subClient.Type.Name;
+                var libraryName = context.DefaultLibraryName;
+                var methodName = subClientName.StartsWith(libraryName)
+                    ? subClientName[libraryName.Length..]
+                    : subClientName;
+
+                var methodSignature = new MethodSignature($"Get{methodName}{ClientBuilder.GetClientSuffix(context)}", $"Initializes a new instance of {subClient.Type.Name}", "public virtual", subClient.Type, null, methodParameters.ToArray());
```

Comment:
I still don't like the way we guess about the name of the method.
Maybe it will work for now, and later we can implement https://github.com/Azure/autorest.csharp/issues/1600 and users will be able to change the name of the method.